internal/logger/logger.go

```diff
@@ -17,32 +17,66 @@ package logger
 import (
 	"io"
 	"log"
+	"log/syslog"
 	"os"
 
 	"github.com/jacobsa/daemonize"
 )
 
+// Syslog file contains logs from all different programmes running on the VM.
+// ProgrammeName is prefixed to all the logs written to syslog. This constant is
+// used to filter the logs from syslog and write it to respective log files -
+// gcsfuse.log in case of GCSFuse.
+const ProgrammeName string = "gcsfuse"
+const GCSFuseAsBackground string = "GCSFUSE_AS_BACKGROUND"
+
 var (
 	defaultLoggerFactory *loggerFactory
 	defaultInfoLogger    *log.Logger
 )
 
 // InitLogFile initializes the logger factory to create loggers that print to
 // a log file.
+// In case of empty file, it starts writing the log to syslog file, which
+// is eventually filtered and redirected to a fixed location using syslog
+// config.
+// Here, background true means, this InitLogFile has been called for the
+// background daemon.
 func InitLogFile(filename string, format string) error {
-	f, err := os.OpenFile(
-		filename,
-		os.O_WRONLY|os.O_CREATE|os.O_APPEND,
-		0644,
-	)
-	if err != nil {
-		return err
+	var f *os.File
+	var sysWriter *syslog.Writer
+	var err error
+	if filename != "" {
+		f, err = os.OpenFile(
+			filename,
+			os.O_WRONLY|os.O_CREATE|os.O_APPEND,
+			0644,
+		)
+		if err != nil {
+			return err
+		}
+	} else {
+		if _, ok := os.LookupEnv(GCSFuseAsBackground); ok {
+			// Priority consist of facility and severity, here facility to specify the
+			// type of system that is logging the message to syslog and severity is log-level.
+			// User applications are allowed to take facility value between LOG_LOCAL0
+			// to LOG_LOCAL7. We are using LOG_LOCAL7 as facility and LOG_DEBUG to write
+			// debug messages.
+			sysWriter, err = syslog.New(syslog.LOG_LOCAL7|syslog.LOG_DEBUG, ProgrammeName)
+			if err != nil {
+				Infof("error while creating sys writer: %v", err)
+				sysWriter = nil
```

Comment:
why do you need to set it to nil?

Reply:
Added just because of safety, since I din't find any doc/comment related to this which says in case of error, writer will be nil.

Removing this.